src/middleware/isLoggedIn.ts

```diff
@@ -1,8 +1,21 @@
 import type { NextFunction, Response, Request } from 'express';
+import * as db from '../db';
+import type { PostgresJsDatabase } from 'drizzle-orm/postgres-js';
+import { eq } from 'drizzle-orm';
 
-export function isLoggedIn() {
-  return function (req: Request, res: Response, next: NextFunction) {
+export function isLoggedIn(dbPool: PostgresJsDatabase<typeof db>) {
+  return async function (req: Request, res: Response, next: NextFunction) {
     if (req.session?.userId) {
+      const rows = await dbPool
+        .select()
+        .from(db.users)
+        .where(eq(db.users.id, req.session.userId))
+        .limit(1);
```

Comment:
I am really not a fan of the `limit` function but I guess in this context its okey as we are only expecting the same user_id. However, in raw sql I would do something like this:

```
SELECT DISTINCT userId
FROM users 
WHERE userId = userId 
```

Using distinct returns only one row if there are multiple duplicates. If it would not then this would be a problem. Even if there are distinct users for some reason then limit would not catch that. However, I understand that this is not easy to do in drizzle so its fine.

Reply:
implemented